cmake/KokkosConfig.cmake.in

```diff
@@ -12,3 +12,17 @@ GET_FILENAME_COMPONENT(Kokkos_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
 INCLUDE("${Kokkos_CMAKE_DIR}/KokkosTargets.cmake")
 INCLUDE("${Kokkos_CMAKE_DIR}/KokkosConfigCommon.cmake")
 UNSET(Kokkos_CMAKE_DIR)
+
+# if CUDA was enabled and separable compilation was specified, e.g.
+#   find_package(Kokkos COMPONENTS separable_compilation)
+# then we set the RULE_LAUNCH_COMPILE and RULE_LAUNCH_LINK
+IF(@Kokkos_ENABLE_CUDA@ AND NOT "separable_compilation" IN_LIST Kokkos_FIND_COMPONENTS)
```

Comment:
I'm going back and forth on what the default mode should be. I'm wondering if it might be better to reverse this - make the person ask for an `NVCC_LAUNCHER` component to globally apply and make the default that no wrapper is added.

separable_compilation is a bit confusing on second read, although I can't think of a clearly better name at the moment.
separable_compilation doesn't seem very descriptive to me on second read.

Reply:
IMO, I think the default should be the way it is because this makes kokkos + cuda and cmake act a lot more like a traditional project. With this setup, a novice to kokkos that doesn't know anything about nvcc_wrapper (or somebody that hasn't read the docs) and thus doesn't set the compiler at all, can just run `ccmake ... && make install` and then add a `find_package(Kokkos)` to their project in blissful ignorance of `nvcc_wrapper` bc it will (hopefully) _just work_. Any project that doesn't want to allow this, can just set a local variable in their CMake code before the find_package and their users won't be able to override it.

Reply:
I chose `separable_compilation` bc:

- We are effectively enabling files designated as the same language in CMake to be sent to separate compilers
- This option is specific to CUDA + NVCC which has a concept of separable compilation so the name will potentially be easy to associate and remember